coil-network-ktor3/src/commonMain/kotlin/coil3/network/ktor3/KtorNetworkFetcher.kt

```diff
@@ -32,7 +32,7 @@ fun KtorNetworkFetcherFactory(
 
 @JvmName("factory")
 fun KtorNetworkFetcherFactory(
-    httpClient: () -> HttpClient,
+    httpClient: () -> HttpClient = { HttpClient() },
```

Comment:
This breaks trailing lambda usage (e.g. `KtorNetworkFetcherFactory { HttpClient() }`), was it intended?
```
Overload resolution ambiguity between candidates:
fun KtorNetworkFetcherFactory(httpClient: () -> HttpClient): NetworkFetcher.Factory
fun KtorNetworkFetcherFactory(httpClient: () -> HttpClient = ..., cacheStrategy: () -> CacheStrategy = ..., connectivityChecker: (Context) -> ConnectivityChecker = ...): NetworkFetcher.Factory
```